docs/sources/k6/next/reference/versioning-and-stability-guarantees.md

```diff
@@ -0,0 +1,285 @@
+
+- All the newly introduced breaking changes would be available under the `v2.0.0-next` version.
+- All the backward-compatible changes would be available under `v1.4.0-next` or `v1.3.6-next`.
+
+### Frequency
+
+The release schedule followed by the new k6 versions:
+
+| Version | Cadence |
+| ------- | ------- |
+| Major   | Once per year |
+| Minor   | Every six weeks |
+| Patch   | As needed, based on priorities |
+
+In alignment with the practice observed in the Grafana ecosystem, they affords us to maintain a regular rhythm, offering opportunities for better feature planning, communication and visibility around releases, and regular cleanup of deprecated functionalities, helping reduce technical debt over time.
+
+## What is our stable and supported API surface?
+
+In defining the stable API surface for k6, we aim to provide clear and stable interfaces guaranteed to be maintained and supported according to our [versioning strategy](#versioning-strategy). Our stable and supported API surface includes several key components, each critical for our users and their integration workflows.
+
+Any change to one of the APIs described below will be considered breaking. Note that the following list is intended as an allowlist, and any APIs, libraries, or components not explicitly mentioned are considered excluded. For example, the k6 REST API is an example of an accessible API not covered by the supported API surface.
+
+### Options & Command-line interface
+
+Our stable and supported API surface includes the k6 options set, as exposed to users via the various configuration methods k6 supports: the options object, config file, environment variables, etc.
+
+The k6 command-line interface (CLI), including commands, subcommands, and flags, is also part of the stable and supported API. These commands are essential for initiating and managing test executions locally and in the cloud.
+
+### JavaScript interpreter
+
+With k6 scripts written in JavaScript and the environment in which they are executed supporting a specific set of JavaScript syntax and ECMAScript features, the k6 JavaScript interpreter is considered part of the k6 stable and supported API surface.
+
+This ensures that users can write and execute scripts with predictable behavior and compatibility over the lifespan of their major version of k6.
+
+### Modules
+
+The *“core”* modules exposed directly to a script by k6, for example, `k6`, `k6/http`, `k6/execution`, are another critical aspect of the k6 experience and of the interface between k6 and its users. As such, they're part of the stable and supported API surface.
+It includes any class, properties, methods, function, or relative signature that is part of the module and is publicly exposed and documented.
+
+The full list of the `core` modules: `k6`, `k6/browser`, `k6/crypto`, `k6/crypto/x509`, `k6/data`, `k6/encoding`, `k6/execution`,  `k6/http`, `k6/metrics`, `k6/net/grpc`, `k6/timers`, `k6/ws`.
```

Comment:
What's the purpose of this "exhaustive" list? What's the plan to keep it up to date? For instance, now there's also `k6/secrets`.

Reply:
```suggestion
```